src/Symfony/Component/ObjectMapper/Metadata/ReflectionObjectMapperMetadataFactory.php

```diff
@@ -21,17 +21,32 @@
  */
 final class ReflectionObjectMapperMetadataFactory implements ObjectMapperMetadataFactoryInterface
 {
+    private ?\SplObjectStorage $reflectionClassCache = null;
+    private array $attributesCache = [];
+
+    public function __construct()
+    {
+        $this->reflectionClassCache = new \SplObjectStorage();
+    }
+
     public function create(object $object, ?string $property = null, array $context = []): array
     {
         try {
-            $refl = new \ReflectionClass($object);
-            $mapTo = [];
-            foreach (($property ? $refl->getProperty($property) : $refl)->getAttributes(Map::class, \ReflectionAttribute::IS_INSTANCEOF) as $mapAttribute) {
-                $map = $mapAttribute->newInstance();
-                $mapTo[] = new Mapping($map->target, $map->source, $map->if, $map->transform);
+            $key = $object::class.($property ?? '');
+
+            if (isset($this->attributesCache[$key])) {
+                return $this->attributesCache[$key];
+            }
+
+            $refl = $this->reflectionClassCache[$object] ??= new \ReflectionClass($object);
```

Comment:
The `SplObjectStorage` will hold a reference to the objects used as key. Can we use a simple array using the class name as the key?

```suggestion
            $refl = $this->reflectionClassCache[$object::class] ??= new \ReflectionClass($object);
```

Reply:
Ah yes, better indeed